inc/classes/class_db_mysql.php

```diff
@@ -2,48 +2,30 @@
 
 class db
 {
-    public $mysqli = 0;
     public $link_id = 0;
     public $query_id = 0;
-    public $record   = array();
-    public $print_sql_error;
+    public $record = array();
     public $success = false;
     public $count_query = 0;
     public $errors = '';
     public $errorsFound = 0;
     public $connectfailure = 0;  //0= no error, 1=connection error, 2=database error
     public $QueryArgs = array();
 
-  // Construktor
-    public function __construct()
```

Comment:
This is true. There are two options:
1. Introduce the constructor again and implement the HTML-Error via echo
2. Document the needed mysqli extension and let it fail with a Fatal error: undefined function

I would go for option 2. Because in the installation process has a check already in it if mysqli is supported. Further more for existing installations i would implement a small /health.php check or whatever that checks all requirements at once.

I would not suggest option 1, because if we want to make lansuite templatable / translatable, we need access to the template and translation system into the database class. This makes it hard to test.
An alternative would be throwing an exception in the constructor. And handle it everywhere.
But i would go for a global system check instead of firing this check at _every_ page call.
What do you prefer?